<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Seungjun Ahn" />


<title>An Introduction to SOHPIE</title>

<script>// Hide empty <a> tag within highlighted CodeBlock for screen reader accessibility (see https://github.com/jgm/pandoc/issues/6352#issuecomment-626106786) -->
// v0.0.1
// Written by JooYoung Seo (jooyoung@psu.edu) and Atsushi Yasumoto on June 1st, 2020.

document.addEventListener('DOMContentLoaded', function() {
  const codeList = document.getElementsByClassName("sourceCode");
  for (var i = 0; i < codeList.length; i++) {
    var linkList = codeList[i].getElementsByTagName('a');
    for (var j = 0; j < linkList.length; j++) {
      if (linkList[j].innerHTML === "") {
        linkList[j].setAttribute('aria-hidden', 'true');
      }
    }
  }
});
</script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">An Introduction to SOHPIE</h1>
<h4 class="author">Seungjun Ahn</h4>
<h4 class="date">October 20th, 2023</h4>



<p>We introduce a Statistical Approach via Pseudo-value Information and Estimation (SOHPIE; pronounced as “Sofie”). This is a regression modeling method for differential network (DN) analysis that can include covariate information in analyzing microbiome data. SOHPIE is a software extension of our methodology work [1].</p>
<div id="requirements" class="section level2">
<h2>Requirements</h2>
<p>Please install these R packages prior to use SOHPIE.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="co"># library(robustbase) # To fit a robust regression.</span></span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="co"># library(parallel) # To use mclapply() when reestimating the association matrix.</span></span>
<span id="cb1-3"><a href="#cb1-3"></a><span class="co"># library(dplyr)  # For the convenience of tabulating p-values, coefficients, and q-values.</span></span>
<span id="cb1-4"><a href="#cb1-4"></a><span class="co"># library(fdrtool) # For false discovery rate control.</span></span>
<span id="cb1-5"><a href="#cb1-5"></a><span class="co"># library(gtools) # To estimate an association matrix via SparCC.</span></span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a><span class="kw">library</span>(SOHPIE)</span></code></pre></div>
<div id="load-the-study-data-from-sohpie-r-package" class="section level3">
<h3>Load the study data from SOHPIE R package:</h3>
<p>Two sample datasets are available in this package. One (<code>combinedamgut</code>) is from the American Gut Project [2] and contains 138 taxa and 268 subjects. In this vignette, the first 30 out of 138 taxa will be used for the simple demonstration purpose. The other (<code>combineddietswap</code>) is from the geographical epidemiology study of diet swap intervention [3] that includes 112 taxa with 37 subjects (20 African Americans from Pittsburgh and 17 rural South Africans). The full data of each study are available in the SpiecEasi and microbiome R packages, respectively.</p>
</div>
<div id="example-i-american-gut-project-data" class="section level3">
<h3>Example I: American Gut Project Data</h3>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a><span class="kw">set.seed</span>(<span class="dv">20050505</span>)</span>
<span id="cb3-2"><a href="#cb3-2"></a><span class="kw">data</span>(combinedamgut) <span class="co"># A complete data containing columns with taxa and clinical covariates.</span></span></code></pre></div>
</div>
<div id="data-processing-for-the-toy-example-using-sample-dataset-from-american-gut-project" class="section level3">
<h3>Data processing for the toy example using sample dataset from American Gut Project:</h3>
<p>The main grouping variable will be the indicator variable for the status of living with a dog. After the data processing, the indices of subjects will be available for each ‘Not living with a dog (Group A)’ vs. ‘Living with a dog (Group B).’ We need these indices for the estimation of group-specific <span class="math inline">\(p \times p\)</span> association matrices (and re-estimation of association matrices for pseudo-value calculations later).</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a><span class="co"># Note: Again, we will use a toy example with the first 30 out of 138 taxa.</span></span>
<span id="cb4-2"><a href="#cb4-2"></a>OTUtab =<span class="st"> </span>combinedamgut[ , <span class="dv">8</span><span class="op">:</span><span class="dv">37</span>]</span>
<span id="cb4-3"><a href="#cb4-3"></a></span>
<span id="cb4-4"><a href="#cb4-4"></a><span class="co"># Clinical/demographic covariates (phenotypic data):</span></span>
<span id="cb4-5"><a href="#cb4-5"></a><span class="co"># Note: All of these covariates in phenodat below will be included in the regression </span></span>
<span id="cb4-6"><a href="#cb4-6"></a><span class="co">#       when you use SOHPIE_DNA function later. Please make sure </span></span>
<span id="cb4-7"><a href="#cb4-7"></a><span class="co">#       phenodat below include variables that will be analyzed only.</span></span>
<span id="cb4-8"><a href="#cb4-8"></a>phenodat =<span class="st"> </span>combinedamgut[, <span class="dv">1</span><span class="op">:</span><span class="dv">7</span>] <span class="co"># first column is ID, so not using it.</span></span></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a><span class="co"># Obtain indices of each grouping factor.</span></span>
<span id="cb5-2"><a href="#cb5-2"></a><span class="co"># In this example, a variable indicating the status of living with a dog was chosen (i.e. bin_dog).</span></span>
<span id="cb5-3"><a href="#cb5-3"></a><span class="co"># Accordingly, Groups A and B imply living without and with a dog, respectively.</span></span>
<span id="cb5-4"><a href="#cb5-4"></a>newindex_grpA =<span class="st"> </span><span class="kw">which</span>(combinedamgut<span class="op">$</span>bin_dog <span class="op">==</span><span class="st"> </span><span class="dv">0</span>)</span>
<span id="cb5-5"><a href="#cb5-5"></a>newindex_grpB =<span class="st"> </span><span class="kw">which</span>(combinedamgut<span class="op">$</span>bin_dog <span class="op">==</span><span class="st"> </span><span class="dv">1</span>)</span></code></pre></div>
</div>
<div id="fit-a-pseudo-value-regression-via-sohpie_dna-function" class="section level3">
<h3>Fit a pseudo-value regression via SOHPIE_DNA() function:</h3>
<p>Upon our data processing step above is complete, we can then fit a pseudo-value regression using <code>SOHPIE_DNA</code> function. An important note! Please provide the object name of each OTU table and clinical/demographic data (i.e. metadata) separately in the function. In addition, you must indicate the object names of the indices for each group of a binary indicator variable that is used as a main predictor variable (e.g. living with a dog vs. without a dog). Lastly, you must enter a trimming proportion <code>c</code>, which ranges from 0.5 to 1.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1"></a>SOHPIEres &lt;-<span class="st"> </span><span class="kw">SOHPIE_DNA</span>(<span class="dt">OTUdat =</span> OTUtab, <span class="dt">clindat =</span> phenodat, </span>
<span id="cb6-2"><a href="#cb6-2"></a>                        <span class="dt">groupA =</span> newindex_grpA, <span class="dt">groupB =</span> newindex_grpB, <span class="dt">c =</span> <span class="fl">0.5</span>)</span></code></pre></div>
</div>
<div id="additional-features-available-in-sohpie-package" class="section level3">
<h3>Additional features available in SOHPIE package:</h3>
<p>Now, I would like to show you that SOHPIE has some convenient tools/functions after fitting a pseudo-value regression. There are functions that you can quickly extract names of taxa that are significantly differentially connected (DC; <code>DCtaxa_tab</code>), as well as p-values (<code>pval</code> and <code>pval_specific_var</code>), adjusted p-values (q-values; <code>qval</code> and <code>qval_specific_var</code>), coefficient estimates (<code>coeff</code> and <code>coeff_specific_var</code>), and standard errors (<code>stderrs</code> and <code>stderrs_specific_var</code>) of all variables that are considered in the regression or a specific variable.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a><span class="co"># qval() function will get you a table with q-values.</span></span>
<span id="cb7-2"><a href="#cb7-2"></a><span class="kw">qval</span>(SOHPIEres)</span>
<span id="cb7-3"><a href="#cb7-3"></a><span class="co">#&gt;        bin_dog          age       sex bin_floss bin_exercise cat_alcohol1</span></span>
<span id="cb7-4"><a href="#cb7-4"></a><span class="co">#&gt; 326792       1 0.3335066191 0.3891881 0.6085508 0.7029038265   0.58249028</span></span>
<span id="cb7-5"><a href="#cb7-5"></a><span class="co">#&gt; 348374       1 0.4959918786 0.4553286 0.6563176 0.0007978104   0.67793327</span></span>
<span id="cb7-6"><a href="#cb7-6"></a><span class="co">#&gt; 181016       1 0.1034221221 0.3437233 0.4613268 0.7906962897   0.91583955</span></span>
<span id="cb7-7"><a href="#cb7-7"></a><span class="co">#&gt; 191687       1 0.1061624991 0.4121733 0.7620091 0.6509072393   0.63046991</span></span>
<span id="cb7-8"><a href="#cb7-8"></a><span class="co">#&gt; 305760       1 0.0623960296 0.3443704 0.3380110 0.2961143241   0.89650834</span></span>
<span id="cb7-9"><a href="#cb7-9"></a><span class="co">#&gt; 326977       1 0.0816294220 0.4511363 0.8534582 0.7191285464   0.85698753</span></span>
<span id="cb7-10"><a href="#cb7-10"></a><span class="co">#&gt; 194648       1 0.1738269828 0.2574068 0.3337861 0.6925366825   0.88023775</span></span>
<span id="cb7-11"><a href="#cb7-11"></a><span class="co">#&gt; 28186        1 0.0007831936 0.2574068 0.7885174 0.8453179274   0.89192131</span></span>
<span id="cb7-12"><a href="#cb7-12"></a><span class="co">#&gt; 541301       1 0.5657343541 0.2574068 0.6908599 0.7506372978   0.90966387</span></span>
<span id="cb7-13"><a href="#cb7-13"></a><span class="co">#&gt; 198941       1 0.1013064347 0.2574068 0.7776566 0.7026636239   0.01754703</span></span>
<span id="cb7-14"><a href="#cb7-14"></a><span class="co">#&gt; 353985       1 0.3223166965 0.3746422 0.6595031 0.6871464142   0.81861042</span></span>
<span id="cb7-15"><a href="#cb7-15"></a><span class="co">#&gt; 187524       1 0.2946551331 0.3593480 0.8264901 0.8042761312   0.85349403</span></span>
<span id="cb7-16"><a href="#cb7-16"></a><span class="co">#&gt; 182054       1 0.4039503606 0.3070761 0.8275860 0.1490472935   0.73146229</span></span>
<span id="cb7-17"><a href="#cb7-17"></a><span class="co">#&gt; 175537       1 0.4490265917 0.2711891 0.5726527 0.8033774242   0.91236812</span></span>
<span id="cb7-18"><a href="#cb7-18"></a><span class="co">#&gt; 9753         1 0.4910740372 0.3911474 0.2141110 0.8025621972   0.90363709</span></span>
<span id="cb7-19"><a href="#cb7-19"></a><span class="co">#&gt; 194211       1 0.5115365048 0.2574068 0.7307118 0.8373385238   0.87655047</span></span>
<span id="cb7-20"><a href="#cb7-20"></a><span class="co">#&gt; 188518       1 0.0151224058 0.3347717 0.7828179 0.3053262763   0.57247901</span></span>
<span id="cb7-21"><a href="#cb7-21"></a><span class="co">#&gt; 189396       1 0.4212166815 0.3494026 0.3350469 0.6958245850   0.62345510</span></span>
<span id="cb7-22"><a href="#cb7-22"></a><span class="co">#&gt; 90487        1 0.0155838264 0.2574068 0.5957072 0.6334803310   0.92405775</span></span>
<span id="cb7-23"><a href="#cb7-23"></a><span class="co">#&gt; 203708       1 0.4307922980 0.3876823 0.8196181 0.4291603609   0.90042477</span></span>
<span id="cb7-24"><a href="#cb7-24"></a><span class="co">#&gt; 173965       1 0.5767127939 0.3928536 0.3326346 0.3775387180   0.87458491</span></span>
<span id="cb7-25"><a href="#cb7-25"></a><span class="co">#&gt; 194661       1 0.2655818775 0.4633322 0.2382683 0.8290773188   0.48876124</span></span>
<span id="cb7-26"><a href="#cb7-26"></a><span class="co">#&gt; 512309       1 0.1044861801 0.3375792 0.7154385 0.7790596913   0.88219494</span></span>
<span id="cb7-27"><a href="#cb7-27"></a><span class="co">#&gt; 170124       1 0.5607160782 0.5135918 0.3142403 0.7070289268   0.85300729</span></span>
<span id="cb7-28"><a href="#cb7-28"></a><span class="co">#&gt; 216862       1 0.1033116429 0.2574068 0.8304994 0.6485739027   0.89802021</span></span>
<span id="cb7-29"><a href="#cb7-29"></a><span class="co">#&gt; 352304       1 0.4001572982 0.4448885 0.6941110 0.7547647075   0.91968832</span></span>
<span id="cb7-30"><a href="#cb7-30"></a><span class="co">#&gt; 191306       1 0.2730470071 0.3820290 0.4569466 0.7873666361   0.88495526</span></span>
<span id="cb7-31"><a href="#cb7-31"></a><span class="co">#&gt; 191541       1 0.3346195307 0.3882961 0.8461190 0.7847328718   0.62178911</span></span>
<span id="cb7-32"><a href="#cb7-32"></a><span class="co">#&gt; 191547       1 0.0931888307 0.2574068 0.8288977 0.8324249858   0.89687588</span></span>
<span id="cb7-33"><a href="#cb7-33"></a><span class="co">#&gt; 195493       1 0.3150635107 0.3739925 0.8218145 0.7061505869   0.91477808</span></span>
<span id="cb7-34"><a href="#cb7-34"></a><span class="co">#&gt;        cat_alcohol2 bin_migraine</span></span>
<span id="cb7-35"><a href="#cb7-35"></a><span class="co">#&gt; 326792    0.4470969   0.43366083</span></span>
<span id="cb7-36"><a href="#cb7-36"></a><span class="co">#&gt; 348374    0.6094692   0.34296707</span></span>
<span id="cb7-37"><a href="#cb7-37"></a><span class="co">#&gt; 181016    0.7118308   0.51037157</span></span>
<span id="cb7-38"><a href="#cb7-38"></a><span class="co">#&gt; 191687    0.5331404   0.22406386</span></span>
<span id="cb7-39"><a href="#cb7-39"></a><span class="co">#&gt; 305760    0.7603810   0.47840625</span></span>
<span id="cb7-40"><a href="#cb7-40"></a><span class="co">#&gt; 326977    0.7106559   0.63133361</span></span>
<span id="cb7-41"><a href="#cb7-41"></a><span class="co">#&gt; 194648    0.6460261   0.60245176</span></span>
<span id="cb7-42"><a href="#cb7-42"></a><span class="co">#&gt; 28186     0.5022682   0.44397328</span></span>
<span id="cb7-43"><a href="#cb7-43"></a><span class="co">#&gt; 541301    0.5325693   0.38364557</span></span>
<span id="cb7-44"><a href="#cb7-44"></a><span class="co">#&gt; 198941    0.5323451   0.32301604</span></span>
<span id="cb7-45"><a href="#cb7-45"></a><span class="co">#&gt; 353985    0.5984587   0.42326881</span></span>
<span id="cb7-46"><a href="#cb7-46"></a><span class="co">#&gt; 187524    0.7177693   0.44374322</span></span>
<span id="cb7-47"><a href="#cb7-47"></a><span class="co">#&gt; 182054    0.7264787   0.64520003</span></span>
<span id="cb7-48"><a href="#cb7-48"></a><span class="co">#&gt; 175537    0.5161692   0.43985159</span></span>
<span id="cb7-49"><a href="#cb7-49"></a><span class="co">#&gt; 9753      0.6307723   0.62995026</span></span>
<span id="cb7-50"><a href="#cb7-50"></a><span class="co">#&gt; 194211    0.7641725   0.37619117</span></span>
<span id="cb7-51"><a href="#cb7-51"></a><span class="co">#&gt; 188518    0.4829608   0.23422186</span></span>
<span id="cb7-52"><a href="#cb7-52"></a><span class="co">#&gt; 189396    0.3389945   0.20953056</span></span>
<span id="cb7-53"><a href="#cb7-53"></a><span class="co">#&gt; 90487     0.4802509   0.50174676</span></span>
<span id="cb7-54"><a href="#cb7-54"></a><span class="co">#&gt; 203708    0.7630714   0.59075030</span></span>
<span id="cb7-55"><a href="#cb7-55"></a><span class="co">#&gt; 173965    0.7567017   0.44288581</span></span>
<span id="cb7-56"><a href="#cb7-56"></a><span class="co">#&gt; 194661    0.4980916   0.40289035</span></span>
<span id="cb7-57"><a href="#cb7-57"></a><span class="co">#&gt; 512309    0.5018482   0.41683800</span></span>
<span id="cb7-58"><a href="#cb7-58"></a><span class="co">#&gt; 170124    0.4920228   0.43786678</span></span>
<span id="cb7-59"><a href="#cb7-59"></a><span class="co">#&gt; 216862    0.7201564   0.04967498</span></span>
<span id="cb7-60"><a href="#cb7-60"></a><span class="co">#&gt; 352304    0.7465225   0.44993928</span></span>
<span id="cb7-61"><a href="#cb7-61"></a><span class="co">#&gt; 191306    0.5246934   0.40835019</span></span>
<span id="cb7-62"><a href="#cb7-62"></a><span class="co">#&gt; 191541    0.5935170   0.36978953</span></span>
<span id="cb7-63"><a href="#cb7-63"></a><span class="co">#&gt; 191547    0.7767038   0.53092184</span></span>
<span id="cb7-64"><a href="#cb7-64"></a><span class="co">#&gt; 195493    0.4978749   0.56935890</span></span></code></pre></div>
<p><code>qval_specific_var</code> function will be useful to retrieve the q-values of a specific variable, <code>bin_dog</code> in this example.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1"></a><span class="co"># Create an object to keep the table with q-values.</span></span>
<span id="cb8-2"><a href="#cb8-2"></a>qvaltab &lt;-<span class="st"> </span><span class="kw">qval</span>(SOHPIEres)</span>
<span id="cb8-3"><a href="#cb8-3"></a><span class="co"># Retrieve a vector of q-values for a single variable of interest.</span></span>
<span id="cb8-4"><a href="#cb8-4"></a><span class="kw">qval_specific_var</span>(<span class="dt">qvaltab =</span> qvaltab, <span class="dt">varname =</span> <span class="st">&quot;bin_dog&quot;</span>)</span>
<span id="cb8-5"><a href="#cb8-5"></a><span class="co">#&gt;        bin_dog</span></span>
<span id="cb8-6"><a href="#cb8-6"></a><span class="co">#&gt; 326792       1</span></span>
<span id="cb8-7"><a href="#cb8-7"></a><span class="co">#&gt; 348374       1</span></span>
<span id="cb8-8"><a href="#cb8-8"></a><span class="co">#&gt; 181016       1</span></span>
<span id="cb8-9"><a href="#cb8-9"></a><span class="co">#&gt; 191687       1</span></span>
<span id="cb8-10"><a href="#cb8-10"></a><span class="co">#&gt; 305760       1</span></span>
<span id="cb8-11"><a href="#cb8-11"></a><span class="co">#&gt; 326977       1</span></span>
<span id="cb8-12"><a href="#cb8-12"></a><span class="co">#&gt; 194648       1</span></span>
<span id="cb8-13"><a href="#cb8-13"></a><span class="co">#&gt; 28186        1</span></span>
<span id="cb8-14"><a href="#cb8-14"></a><span class="co">#&gt; 541301       1</span></span>
<span id="cb8-15"><a href="#cb8-15"></a><span class="co">#&gt; 198941       1</span></span>
<span id="cb8-16"><a href="#cb8-16"></a><span class="co">#&gt; 353985       1</span></span>
<span id="cb8-17"><a href="#cb8-17"></a><span class="co">#&gt; 187524       1</span></span>
<span id="cb8-18"><a href="#cb8-18"></a><span class="co">#&gt; 182054       1</span></span>
<span id="cb8-19"><a href="#cb8-19"></a><span class="co">#&gt; 175537       1</span></span>
<span id="cb8-20"><a href="#cb8-20"></a><span class="co">#&gt; 9753         1</span></span>
<span id="cb8-21"><a href="#cb8-21"></a><span class="co">#&gt; 194211       1</span></span>
<span id="cb8-22"><a href="#cb8-22"></a><span class="co">#&gt; 188518       1</span></span>
<span id="cb8-23"><a href="#cb8-23"></a><span class="co">#&gt; 189396       1</span></span>
<span id="cb8-24"><a href="#cb8-24"></a><span class="co">#&gt; 90487        1</span></span>
<span id="cb8-25"><a href="#cb8-25"></a><span class="co">#&gt; 203708       1</span></span>
<span id="cb8-26"><a href="#cb8-26"></a><span class="co">#&gt; 173965       1</span></span>
<span id="cb8-27"><a href="#cb8-27"></a><span class="co">#&gt; 194661       1</span></span>
<span id="cb8-28"><a href="#cb8-28"></a><span class="co">#&gt; 512309       1</span></span>
<span id="cb8-29"><a href="#cb8-29"></a><span class="co">#&gt; 170124       1</span></span>
<span id="cb8-30"><a href="#cb8-30"></a><span class="co">#&gt; 216862       1</span></span>
<span id="cb8-31"><a href="#cb8-31"></a><span class="co">#&gt; 352304       1</span></span>
<span id="cb8-32"><a href="#cb8-32"></a><span class="co">#&gt; 191306       1</span></span>
<span id="cb8-33"><a href="#cb8-33"></a><span class="co">#&gt; 191541       1</span></span>
<span id="cb8-34"><a href="#cb8-34"></a><span class="co">#&gt; 191547       1</span></span>
<span id="cb8-35"><a href="#cb8-35"></a><span class="co">#&gt; 195493       1</span></span></code></pre></div>
<p><code>DCtaxa_tab</code> will return a list containing of (1) names and q-values of taxa that are significantly DC between two biological conditions and (2) names of DC taxa only.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1"></a><span class="co"># Please do NOT forget to provide the name of variable in DCtaxa_tab(groupvar = )</span></span>
<span id="cb9-2"><a href="#cb9-2"></a><span class="co"># and the level of significance (0.3 in this example).</span></span>
<span id="cb9-3"><a href="#cb9-3"></a>DCtaxa_tab &lt;-<span class="st"> </span><span class="kw">DCtaxa_tab</span>(<span class="dt">qvaltab =</span> qvaltab, <span class="dt">groupvar =</span> <span class="st">&quot;bin_dog&quot;</span>, <span class="dt">alpha =</span> <span class="fl">0.3</span>)</span>
<span id="cb9-4"><a href="#cb9-4"></a>DCtaxa_tab</span>
<span id="cb9-5"><a href="#cb9-5"></a><span class="co">#&gt; $DCtaxa_complete_tab</span></span>
<span id="cb9-6"><a href="#cb9-6"></a><span class="co">#&gt; [1] bin_dog</span></span>
<span id="cb9-7"><a href="#cb9-7"></a><span class="co">#&gt; &lt;0 rows&gt; (or 0-length row.names)</span></span>
<span id="cb9-8"><a href="#cb9-8"></a><span class="co">#&gt; </span></span>
<span id="cb9-9"><a href="#cb9-9"></a><span class="co">#&gt; $DCtaxa_names_only</span></span>
<span id="cb9-10"><a href="#cb9-10"></a><span class="co">#&gt; character(0)</span></span></code></pre></div>
</div>
<div id="example-ii-diet-exchange-study-data" class="section level3">
<h3>Example II: Diet Exchange Study Data</h3>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1"></a><span class="kw">data</span>(combineddietswap)</span>
<span id="cb10-2"><a href="#cb10-2"></a></span>
<span id="cb10-3"><a href="#cb10-3"></a>OTUtab =<span class="st"> </span>combineddietswap[ , <span class="dv">5</span><span class="op">:</span><span class="kw">ncol</span>(combineddietswap)]</span>
<span id="cb10-4"><a href="#cb10-4"></a>phenodat =<span class="st"> </span>combineddietswap[ , <span class="dv">1</span><span class="op">:</span><span class="dv">4</span>] <span class="co"># first column is ID, so not using it.</span></span></code></pre></div>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1"></a><span class="co"># Obtain indices for each groups </span></span>
<span id="cb11-2"><a href="#cb11-2"></a><span class="co"># (i.e. African-Americans from Pittsburgh (AAM) vs. Africans from rural South Africa (AFR))</span></span>
<span id="cb11-3"><a href="#cb11-3"></a><span class="co"># at baseline (time = 1) and at 29-days (time = 6)</span></span>
<span id="cb11-4"><a href="#cb11-4"></a></span>
<span id="cb11-5"><a href="#cb11-5"></a><span class="co"># Group A1 for AAM at baseline.</span></span>
<span id="cb11-6"><a href="#cb11-6"></a>newindex_A1 =<span class="st"> </span><span class="kw">which</span>(combineddietswap<span class="op">$</span>timepoint <span class="op">==</span><span class="st"> </span><span class="dv">1</span> <span class="op">&amp;</span><span class="st"> </span>combineddietswap<span class="op">$</span>nationality <span class="op">==</span><span class="st"> &quot;AAM&quot;</span>)</span>
<span id="cb11-7"><a href="#cb11-7"></a><span class="co"># Group A6 for AAM at 29-days.</span></span>
<span id="cb11-8"><a href="#cb11-8"></a>newindex_A6 =<span class="st"> </span><span class="kw">which</span>(combineddietswap<span class="op">$</span>timepoint <span class="op">==</span><span class="st"> </span><span class="dv">6</span> <span class="op">&amp;</span><span class="st"> </span>combineddietswap<span class="op">$</span>nationality <span class="op">==</span><span class="st"> &quot;AAM&quot;</span>)</span>
<span id="cb11-9"><a href="#cb11-9"></a><span class="co"># Group B1 for AFR at baseline.</span></span>
<span id="cb11-10"><a href="#cb11-10"></a>newindex_B1 =<span class="st"> </span><span class="kw">which</span>(combineddietswap<span class="op">$</span>timepoint <span class="op">==</span><span class="st"> </span><span class="dv">1</span> <span class="op">&amp;</span><span class="st"> </span>combineddietswap<span class="op">$</span>nationality <span class="op">==</span><span class="st"> &quot;AFR&quot;</span>)</span>
<span id="cb11-11"><a href="#cb11-11"></a><span class="co"># Group A6 for AFR at 29-days.</span></span>
<span id="cb11-12"><a href="#cb11-12"></a>newindex_B6 =<span class="st"> </span><span class="kw">which</span>(combineddietswap<span class="op">$</span>timepoint <span class="op">==</span><span class="st"> </span><span class="dv">6</span> <span class="op">&amp;</span><span class="st"> </span>combineddietswap<span class="op">$</span>nationality <span class="op">==</span><span class="st"> &quot;AFR&quot;</span>)</span></code></pre></div>
<p>We are done with loading the data and obtaining indices for each group for each time point. We then move onto the analysis step!</p>
<p>The first step is to estimate and re-estimate association matrices for each of groups A1, A6, B1, and A6 shown above. Of note, a list output comprises <code>data.frame</code> objects as <code>assomat</code> and <code>reest.assomat</code>.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1"></a>est_asso_matA1 =<span class="st"> </span><span class="kw">asso_mat</span>(<span class="dt">OTUdat=</span>OTUtab, <span class="dt">group=</span>newindex_A1)</span>
<span id="cb12-2"><a href="#cb12-2"></a>est_asso_matA6 =<span class="st"> </span><span class="kw">asso_mat</span>(<span class="dt">OTUdat=</span>OTUtab, <span class="dt">group=</span>newindex_A6)</span>
<span id="cb12-3"><a href="#cb12-3"></a>est_asso_matB1 =<span class="st"> </span><span class="kw">asso_mat</span>(<span class="dt">OTUdat=</span>OTUtab, <span class="dt">group=</span>newindex_B1)</span>
<span id="cb12-4"><a href="#cb12-4"></a>est_asso_matB6 =<span class="st"> </span><span class="kw">asso_mat</span>(<span class="dt">OTUdat=</span>OTUtab, <span class="dt">group=</span>newindex_B6)</span>
<span id="cb12-5"><a href="#cb12-5"></a></span>
<span id="cb12-6"><a href="#cb12-6"></a><span class="co">## For each group, we take the difference of estimated </span></span>
<span id="cb12-7"><a href="#cb12-7"></a><span class="co">## association matrices between time points (29-days minus baseline).</span></span>
<span id="cb12-8"><a href="#cb12-8"></a>asso_mat_diffA61 =<span class="st"> </span>est_asso_matA6<span class="op">$</span>assomat <span class="op">-</span><span class="st"> </span>est_asso_matA1<span class="op">$</span>assomat</span>
<span id="cb12-9"><a href="#cb12-9"></a>asso_mat_diffB61 =<span class="st"> </span>est_asso_matB6<span class="op">$</span>assomat <span class="op">-</span><span class="st"> </span>est_asso_matB1<span class="op">$</span>assomat</span>
<span id="cb12-10"><a href="#cb12-10"></a></span>
<span id="cb12-11"><a href="#cb12-11"></a><span class="co">## Similarly, for each group, we take the difference of re-estimated </span></span>
<span id="cb12-12"><a href="#cb12-12"></a><span class="co">## association matrices between time points (29-days minus baseline).</span></span>
<span id="cb12-13"><a href="#cb12-13"></a>asso_mat_drop_diffA61 =<span class="st"> </span><span class="kw">mapply</span>(<span class="st">&#39;-&#39;</span>, est_asso_matA6<span class="op">$</span>reest.assomat,</span>
<span id="cb12-14"><a href="#cb12-14"></a>                               est_asso_matA1<span class="op">$</span>reest.assomat, <span class="dt">SIMPLIFY=</span><span class="ot">FALSE</span>)</span>
<span id="cb12-15"><a href="#cb12-15"></a>asso_mat_drop_diffB61 =<span class="st"> </span><span class="kw">mapply</span>(<span class="st">&#39;-&#39;</span>, est_asso_matB6<span class="op">$</span>reest.assomat,</span>
<span id="cb12-16"><a href="#cb12-16"></a>                               est_asso_matB1<span class="op">$</span>reest.assomat, <span class="dt">SIMPLIFY=</span><span class="ot">FALSE</span>)</span></code></pre></div>
<p>Then, for each group, we calculate changes in network centrality of each taxa between the association matrices estimated from the whole data and between the association matrices re-estimated from the leave-one-out sample.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1"></a><span class="co">## For changes in network centrality between association matrices estimated from the whole data.</span></span>
<span id="cb13-2"><a href="#cb13-2"></a>thetahat_grpA =<span class="st"> </span><span class="kw">thetahats</span>(asso_mat_diffA61)</span>
<span id="cb13-3"><a href="#cb13-3"></a>thetahat_grpB =<span class="st"> </span><span class="kw">thetahats</span>(asso_mat_diffB61)</span>
<span id="cb13-4"><a href="#cb13-4"></a><span class="co">## For changes in network centrality between association matrices </span></span>
<span id="cb13-5"><a href="#cb13-5"></a><span class="co">## re-estimated from the leave-one-out sample.</span></span>
<span id="cb13-6"><a href="#cb13-6"></a>thetahat_drop_grpA =<span class="st"> </span><span class="kw">sapply</span>(asso_mat_drop_diffA61, thetahats)</span>
<span id="cb13-7"><a href="#cb13-7"></a>thetahat_drop_grpB =<span class="st"> </span><span class="kw">sapply</span>(asso_mat_drop_diffB61, thetahats)</span></code></pre></div>
<p>Next, we calculate jackknife pseudo-values.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1"></a><span class="co"># Sample sizes for each group.</span></span>
<span id="cb14-2"><a href="#cb14-2"></a>n_A &lt;-<span class="st"> </span><span class="kw">length</span>(newindex_A1) </span>
<span id="cb14-3"><a href="#cb14-3"></a>n_B &lt;-<span class="st"> </span><span class="kw">length</span>(newindex_B1)</span>
<span id="cb14-4"><a href="#cb14-4"></a></span>
<span id="cb14-5"><a href="#cb14-5"></a><span class="co"># Jackknife pseudo-values.</span></span>
<span id="cb14-6"><a href="#cb14-6"></a>thetatilde_grpA =<span class="st"> </span><span class="kw">thetatildefun</span>(thetahat_grpA, thetahat_drop_grpA, n_A)</span>
<span id="cb14-7"><a href="#cb14-7"></a>thetatilde_grpB =<span class="st"> </span><span class="kw">thetatildefun</span>(thetahat_grpB, thetahat_drop_grpB, n_B)</span>
<span id="cb14-8"><a href="#cb14-8"></a></span>
<span id="cb14-9"><a href="#cb14-9"></a>thetatilde =<span class="st"> </span><span class="kw">rbind</span>(thetatilde_grpA, thetatilde_grpB)</span></code></pre></div>
<p>Fit a robust regression regressing covariates on pseudo-values. Please note that the metadata/phenotype data should contain a set of predictors to be fitted only, so choose them wisely. Additionally, a trimming proportion <code>c</code> of the least trimmed squares estimator should be entered between 0.5 and 1.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1"></a>fitmod =<span class="st"> </span><span class="kw">pseudoreg</span>(<span class="dt">pseudoval=</span>thetatilde, <span class="dt">clindat=</span>phenodat, <span class="dt">c=</span><span class="fl">0.5</span>)</span></code></pre></div>
<p>Lastly, we can extract summary results from the fitted model from <code>fitmod</code> object above. A list of <code>data.frame</code> objects for coefficient estimates, p-values, and q-values will be available.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1"></a>summary.result =<span class="st"> </span><span class="kw">pseudoreg.summary</span>(<span class="dt">pseudo.reg.res=</span>fitmod, </span>
<span id="cb16-2"><a href="#cb16-2"></a>                                   <span class="dt">taxanames=</span><span class="kw">colnames</span>(OTUtab))</span>
<span id="cb16-3"><a href="#cb16-3"></a></span>
<span id="cb16-4"><a href="#cb16-4"></a><span class="co">## In this study, the main grouping variable was nationality. </span></span>
<span id="cb16-5"><a href="#cb16-5"></a><span class="co"># We can use qval_specific_var to see the q-values only.</span></span>
<span id="cb16-6"><a href="#cb16-6"></a><span class="co"># of nationality. Alternatively, we further use DCtaxa_tab to see</span></span>
<span id="cb16-7"><a href="#cb16-7"></a><span class="co"># the DC taxa with their p-values.</span></span>
<span id="cb16-8"><a href="#cb16-8"></a><span class="kw">qval_specific_var</span>(summary.result<span class="op">$</span>q_values, <span class="dt">varname =</span> <span class="st">&quot;nationalityAFR&quot;</span>)</span>
<span id="cb16-9"><a href="#cb16-9"></a><span class="co">#&gt;                                       nationalityAFR</span></span>
<span id="cb16-10"><a href="#cb16-10"></a><span class="co">#&gt; Akkermansia                              0.656021088</span></span>
<span id="cb16-11"><a href="#cb16-11"></a><span class="co">#&gt; Alcaligenes faecalis et rel.             0.697952103</span></span>
<span id="cb16-12"><a href="#cb16-12"></a><span class="co">#&gt; Allistipes et rel.                       0.757496542</span></span>
<span id="cb16-13"><a href="#cb16-13"></a><span class="co">#&gt; Anaerostipes caccae et rel.              0.419192540</span></span>
<span id="cb16-14"><a href="#cb16-14"></a><span class="co">#&gt; Anaerotruncus colihominis et rel.        0.668645846</span></span>
<span id="cb16-15"><a href="#cb16-15"></a><span class="co">#&gt; Anaerovorax odorimutans et rel.          0.264856929</span></span>
<span id="cb16-16"><a href="#cb16-16"></a><span class="co">#&gt; Aquabacterium                            0.195376727</span></span>
<span id="cb16-17"><a href="#cb16-17"></a><span class="co">#&gt; Atopobium                                0.073176926</span></span>
<span id="cb16-18"><a href="#cb16-18"></a><span class="co">#&gt; Bacillus                                 0.673758100</span></span>
<span id="cb16-19"><a href="#cb16-19"></a><span class="co">#&gt; Bacteroides fragilis et rel.             0.035519751</span></span>
<span id="cb16-20"><a href="#cb16-20"></a><span class="co">#&gt; Bacteroides intestinalis et rel.         0.621518988</span></span>
<span id="cb16-21"><a href="#cb16-21"></a><span class="co">#&gt; Bacteroides ovatus et rel.               0.044822294</span></span>
<span id="cb16-22"><a href="#cb16-22"></a><span class="co">#&gt; Bacteroides plebeius et rel.             0.595063296</span></span>
<span id="cb16-23"><a href="#cb16-23"></a><span class="co">#&gt; Bacteroides splachnicus et rel.          0.672181067</span></span>
<span id="cb16-24"><a href="#cb16-24"></a><span class="co">#&gt; Bacteroides stercoris et rel.            0.650922141</span></span>
<span id="cb16-25"><a href="#cb16-25"></a><span class="co">#&gt; Bacteroides uniformis et rel.            0.138742256</span></span>
<span id="cb16-26"><a href="#cb16-26"></a><span class="co">#&gt; Bacteroides vulgatus et rel.             0.034013932</span></span>
<span id="cb16-27"><a href="#cb16-27"></a><span class="co">#&gt; Bifidobacterium                          0.618351854</span></span>
<span id="cb16-28"><a href="#cb16-28"></a><span class="co">#&gt; Bilophila et rel.                        0.083458213</span></span>
<span id="cb16-29"><a href="#cb16-29"></a><span class="co">#&gt; Brachyspira                              0.712180734</span></span>
<span id="cb16-30"><a href="#cb16-30"></a><span class="co">#&gt; Bryantella formatexigens et rel.         0.509315681</span></span>
<span id="cb16-31"><a href="#cb16-31"></a><span class="co">#&gt; Bulleidia moorei et rel.                 0.371973569</span></span>
<span id="cb16-32"><a href="#cb16-32"></a><span class="co">#&gt; Burkholderia                             0.145675949</span></span>
<span id="cb16-33"><a href="#cb16-33"></a><span class="co">#&gt; Butyrivibrio crossotus et rel.           0.668484659</span></span>
<span id="cb16-34"><a href="#cb16-34"></a><span class="co">#&gt; Campylobacter                            0.711069109</span></span>
<span id="cb16-35"><a href="#cb16-35"></a><span class="co">#&gt; Catenibacterium mitsuokai et rel.        0.582119077</span></span>
<span id="cb16-36"><a href="#cb16-36"></a><span class="co">#&gt; Clostridium (sensu stricto)              0.391640539</span></span>
<span id="cb16-37"><a href="#cb16-37"></a><span class="co">#&gt; Clostridium cellulosi et rel.            0.361169562</span></span>
<span id="cb16-38"><a href="#cb16-38"></a><span class="co">#&gt; Clostridium colinum et rel.              0.314471156</span></span>
<span id="cb16-39"><a href="#cb16-39"></a><span class="co">#&gt; Clostridium difficile et rel.            0.638529824</span></span>
<span id="cb16-40"><a href="#cb16-40"></a><span class="co">#&gt; Clostridium leptum et rel.               0.459619387</span></span>
<span id="cb16-41"><a href="#cb16-41"></a><span class="co">#&gt; Clostridium nexile et rel.               0.517564673</span></span>
<span id="cb16-42"><a href="#cb16-42"></a><span class="co">#&gt; Clostridium orbiscindens et rel.         0.694621918</span></span>
<span id="cb16-43"><a href="#cb16-43"></a><span class="co">#&gt; Clostridium ramosum et rel.              0.204245652</span></span>
<span id="cb16-44"><a href="#cb16-44"></a><span class="co">#&gt; Clostridium sphenoides et rel.           0.728222226</span></span>
<span id="cb16-45"><a href="#cb16-45"></a><span class="co">#&gt; Clostridium stercorarium et rel.         0.761438204</span></span>
<span id="cb16-46"><a href="#cb16-46"></a><span class="co">#&gt; Clostridium symbiosum et rel.            0.531366952</span></span>
<span id="cb16-47"><a href="#cb16-47"></a><span class="co">#&gt; Collinsella                              0.234087001</span></span>
<span id="cb16-48"><a href="#cb16-48"></a><span class="co">#&gt; Coprobacillus catenaformis et rel.       0.001459571</span></span>
<span id="cb16-49"><a href="#cb16-49"></a><span class="co">#&gt; Coprococcus eutactus et rel.             0.299058393</span></span>
<span id="cb16-50"><a href="#cb16-50"></a><span class="co">#&gt; Corynebacterium                          0.237937836</span></span>
<span id="cb16-51"><a href="#cb16-51"></a><span class="co">#&gt; Desulfovibrio et rel.                    0.659353237</span></span>
<span id="cb16-52"><a href="#cb16-52"></a><span class="co">#&gt; Dialister                                0.721195606</span></span>
<span id="cb16-53"><a href="#cb16-53"></a><span class="co">#&gt; Dorea formicigenerans et rel.            0.759582869</span></span>
<span id="cb16-54"><a href="#cb16-54"></a><span class="co">#&gt; Eggerthella lenta et rel.                0.078400654</span></span>
<span id="cb16-55"><a href="#cb16-55"></a><span class="co">#&gt; Enterobacter aerogenes et rel.           0.349241669</span></span>
<span id="cb16-56"><a href="#cb16-56"></a><span class="co">#&gt; Enterococcus                             0.709904370</span></span>
<span id="cb16-57"><a href="#cb16-57"></a><span class="co">#&gt; Escherichia coli et rel.                 0.524142125</span></span>
<span id="cb16-58"><a href="#cb16-58"></a><span class="co">#&gt; Eubacterium biforme et rel.              0.750869548</span></span>
<span id="cb16-59"><a href="#cb16-59"></a><span class="co">#&gt; Eubacterium cylindroides et rel.         0.633016505</span></span>
<span id="cb16-60"><a href="#cb16-60"></a><span class="co">#&gt; Eubacterium hallii et rel.               0.661554759</span></span>
<span id="cb16-61"><a href="#cb16-61"></a><span class="co">#&gt; Eubacterium limosum et rel.              0.663801438</span></span>
<span id="cb16-62"><a href="#cb16-62"></a><span class="co">#&gt; Eubacterium rectale et rel.              0.001048128</span></span>
<span id="cb16-63"><a href="#cb16-63"></a><span class="co">#&gt; Eubacterium siraeum et rel.              0.001048128</span></span>
<span id="cb16-64"><a href="#cb16-64"></a><span class="co">#&gt; Eubacterium ventriosum et rel.           0.128788309</span></span>
<span id="cb16-65"><a href="#cb16-65"></a><span class="co">#&gt; Faecalibacterium prausnitzii et rel.     0.644096129</span></span>
<span id="cb16-66"><a href="#cb16-66"></a><span class="co">#&gt; Fusobacteria                             0.555685439</span></span>
<span id="cb16-67"><a href="#cb16-67"></a><span class="co">#&gt; Haemophilus                              0.622367506</span></span>
<span id="cb16-68"><a href="#cb16-68"></a><span class="co">#&gt; Helicobacter                             0.314897458</span></span>
<span id="cb16-69"><a href="#cb16-69"></a><span class="co">#&gt; Klebisiella pneumoniae et rel.           0.753363113</span></span>
<span id="cb16-70"><a href="#cb16-70"></a><span class="co">#&gt; Lachnobacillus bovis et rel.             0.531631713</span></span>
<span id="cb16-71"><a href="#cb16-71"></a><span class="co">#&gt; Lachnospira pectinoschiza et rel.        0.413071378</span></span>
<span id="cb16-72"><a href="#cb16-72"></a><span class="co">#&gt; Lactobacillus catenaformis et rel.       0.739650350</span></span>
<span id="cb16-73"><a href="#cb16-73"></a><span class="co">#&gt; Lactobacillus gasseri et rel.            0.730725491</span></span>
<span id="cb16-74"><a href="#cb16-74"></a><span class="co">#&gt; Lactobacillus plantarum et rel.          0.185560537</span></span>
<span id="cb16-75"><a href="#cb16-75"></a><span class="co">#&gt; Lactobacillus salivarius et rel.         0.202934379</span></span>
<span id="cb16-76"><a href="#cb16-76"></a><span class="co">#&gt; Lactococcus                              0.001048128</span></span>
<span id="cb16-77"><a href="#cb16-77"></a><span class="co">#&gt; Leminorella                              0.230251123</span></span>
<span id="cb16-78"><a href="#cb16-78"></a><span class="co">#&gt; Megamonas hypermegale et rel.            0.694628028</span></span>
<span id="cb16-79"><a href="#cb16-79"></a><span class="co">#&gt; Megasphaera elsdenii et rel.             0.211992016</span></span>
<span id="cb16-80"><a href="#cb16-80"></a><span class="co">#&gt; Mitsuokella multiacida et rel.           0.729784685</span></span>
<span id="cb16-81"><a href="#cb16-81"></a><span class="co">#&gt; Moraxellaceae                            0.291006041</span></span>
<span id="cb16-82"><a href="#cb16-82"></a><span class="co">#&gt; Oceanospirillum                          0.734683936</span></span>
<span id="cb16-83"><a href="#cb16-83"></a><span class="co">#&gt; Oscillospira guillermondii et rel.       0.683147265</span></span>
<span id="cb16-84"><a href="#cb16-84"></a><span class="co">#&gt; Outgrouping clostridium cluster XIVa     0.465352252</span></span>
<span id="cb16-85"><a href="#cb16-85"></a><span class="co">#&gt; Oxalobacter formigenes et rel.           0.295900369</span></span>
<span id="cb16-86"><a href="#cb16-86"></a><span class="co">#&gt; Papillibacter cinnamivorans et rel.      0.452280674</span></span>
<span id="cb16-87"><a href="#cb16-87"></a><span class="co">#&gt; Parabacteroides distasonis et rel.       0.444258228</span></span>
<span id="cb16-88"><a href="#cb16-88"></a><span class="co">#&gt; Peptococcus niger et rel.                0.748523402</span></span>
<span id="cb16-89"><a href="#cb16-89"></a><span class="co">#&gt; Peptostreptococcus micros et rel.        0.510876670</span></span>
<span id="cb16-90"><a href="#cb16-90"></a><span class="co">#&gt; Phascolarctobacterium faecium et rel.    0.046609160</span></span>
<span id="cb16-91"><a href="#cb16-91"></a><span class="co">#&gt; Prevotella melaninogenica et rel.        0.469565314</span></span>
<span id="cb16-92"><a href="#cb16-92"></a><span class="co">#&gt; Prevotella oralis et rel.                0.477391811</span></span>
<span id="cb16-93"><a href="#cb16-93"></a><span class="co">#&gt; Prevotella ruminicola et rel.            0.586679550</span></span>
<span id="cb16-94"><a href="#cb16-94"></a><span class="co">#&gt; Prevotella tannerae et rel.              0.455649683</span></span>
<span id="cb16-95"><a href="#cb16-95"></a><span class="co">#&gt; Propionibacterium                        0.440568538</span></span>
<span id="cb16-96"><a href="#cb16-96"></a><span class="co">#&gt; Proteus et rel.                          0.267804676</span></span>
<span id="cb16-97"><a href="#cb16-97"></a><span class="co">#&gt; Roseburia intestinalis et rel.           0.711504977</span></span>
<span id="cb16-98"><a href="#cb16-98"></a><span class="co">#&gt; Ruminococcus bromii et rel.              0.226119652</span></span>
<span id="cb16-99"><a href="#cb16-99"></a><span class="co">#&gt; Ruminococcus callidus et rel.            0.733204120</span></span>
<span id="cb16-100"><a href="#cb16-100"></a><span class="co">#&gt; Ruminococcus gnavus et rel.              0.016099651</span></span>
<span id="cb16-101"><a href="#cb16-101"></a><span class="co">#&gt; Ruminococcus lactaris et rel.            0.485267601</span></span>
<span id="cb16-102"><a href="#cb16-102"></a><span class="co">#&gt; Ruminococcus obeum et rel.               0.044620773</span></span>
<span id="cb16-103"><a href="#cb16-103"></a><span class="co">#&gt; Serratia                                 0.497287487</span></span>
<span id="cb16-104"><a href="#cb16-104"></a><span class="co">#&gt; Sporobacter termitidis et rel.           0.503412730</span></span>
<span id="cb16-105"><a href="#cb16-105"></a><span class="co">#&gt; Staphylococcus                           0.050978693</span></span>
<span id="cb16-106"><a href="#cb16-106"></a><span class="co">#&gt; Streptococcus bovis et rel.              0.233413109</span></span>
<span id="cb16-107"><a href="#cb16-107"></a><span class="co">#&gt; Streptococcus intermedius et rel.        0.506043136</span></span>
<span id="cb16-108"><a href="#cb16-108"></a><span class="co">#&gt; Streptococcus mitis et rel.              0.731204594</span></span>
<span id="cb16-109"><a href="#cb16-109"></a><span class="co">#&gt; Subdoligranulum variable at rel.         0.239512504</span></span>
<span id="cb16-110"><a href="#cb16-110"></a><span class="co">#&gt; Sutterella wadsworthia et rel.           0.628461198</span></span>
<span id="cb16-111"><a href="#cb16-111"></a><span class="co">#&gt; Tannerella et rel.                       0.520336541</span></span>
<span id="cb16-112"><a href="#cb16-112"></a><span class="co">#&gt; Uncultured Bacteroidetes                 0.415039103</span></span>
<span id="cb16-113"><a href="#cb16-113"></a><span class="co">#&gt; Uncultured Clostridiales I               0.753587735</span></span>
<span id="cb16-114"><a href="#cb16-114"></a><span class="co">#&gt; Uncultured Clostridiales II              0.202319965</span></span>
<span id="cb16-115"><a href="#cb16-115"></a><span class="co">#&gt; Uncultured Mollicutes                    0.525998562</span></span>
<span id="cb16-116"><a href="#cb16-116"></a><span class="co">#&gt; Uncultured Selenomonadaceae              0.083965037</span></span>
<span id="cb16-117"><a href="#cb16-117"></a><span class="co">#&gt; Veillonella                              0.065112613</span></span>
<span id="cb16-118"><a href="#cb16-118"></a><span class="co">#&gt; Vibrio                                   0.196090749</span></span>
<span id="cb16-119"><a href="#cb16-119"></a><span class="co">#&gt; Weissella et rel.                        0.684806123</span></span>
<span id="cb16-120"><a href="#cb16-120"></a><span class="co">#&gt; Xanthomonadaceae                         0.738193936</span></span>
<span id="cb16-121"><a href="#cb16-121"></a><span class="co">#&gt; Yersinia et rel.                         0.106486396</span></span>
<span id="cb16-122"><a href="#cb16-122"></a><span class="kw">DCtaxa_tab</span>(summary.result<span class="op">$</span>q_values, <span class="dt">groupvar =</span> <span class="st">&quot;nationalityAFR&quot;</span>, <span class="dt">alpha=</span><span class="fl">0.05</span>)</span>
<span id="cb16-123"><a href="#cb16-123"></a><span class="co">#&gt; $DCtaxa_complete_tab</span></span>
<span id="cb16-124"><a href="#cb16-124"></a><span class="co">#&gt;                                       nationalityAFR</span></span>
<span id="cb16-125"><a href="#cb16-125"></a><span class="co">#&gt; Bacteroides fragilis et rel.             0.035519751</span></span>
<span id="cb16-126"><a href="#cb16-126"></a><span class="co">#&gt; Bacteroides ovatus et rel.               0.044822294</span></span>
<span id="cb16-127"><a href="#cb16-127"></a><span class="co">#&gt; Bacteroides vulgatus et rel.             0.034013932</span></span>
<span id="cb16-128"><a href="#cb16-128"></a><span class="co">#&gt; Coprobacillus catenaformis et rel.       0.001459571</span></span>
<span id="cb16-129"><a href="#cb16-129"></a><span class="co">#&gt; Eubacterium rectale et rel.              0.001048128</span></span>
<span id="cb16-130"><a href="#cb16-130"></a><span class="co">#&gt; Eubacterium siraeum et rel.              0.001048128</span></span>
<span id="cb16-131"><a href="#cb16-131"></a><span class="co">#&gt; Lactococcus                              0.001048128</span></span>
<span id="cb16-132"><a href="#cb16-132"></a><span class="co">#&gt; Phascolarctobacterium faecium et rel.    0.046609160</span></span>
<span id="cb16-133"><a href="#cb16-133"></a><span class="co">#&gt; Ruminococcus gnavus et rel.              0.016099651</span></span>
<span id="cb16-134"><a href="#cb16-134"></a><span class="co">#&gt; Ruminococcus obeum et rel.               0.044620773</span></span>
<span id="cb16-135"><a href="#cb16-135"></a><span class="co">#&gt; </span></span>
<span id="cb16-136"><a href="#cb16-136"></a><span class="co">#&gt; $DCtaxa_names_only</span></span>
<span id="cb16-137"><a href="#cb16-137"></a><span class="co">#&gt;  [1] &quot;Bacteroides fragilis et rel.&quot;         </span></span>
<span id="cb16-138"><a href="#cb16-138"></a><span class="co">#&gt;  [2] &quot;Bacteroides ovatus et rel.&quot;           </span></span>
<span id="cb16-139"><a href="#cb16-139"></a><span class="co">#&gt;  [3] &quot;Bacteroides vulgatus et rel.&quot;         </span></span>
<span id="cb16-140"><a href="#cb16-140"></a><span class="co">#&gt;  [4] &quot;Coprobacillus catenaformis et rel.&quot;   </span></span>
<span id="cb16-141"><a href="#cb16-141"></a><span class="co">#&gt;  [5] &quot;Eubacterium rectale et rel.&quot;          </span></span>
<span id="cb16-142"><a href="#cb16-142"></a><span class="co">#&gt;  [6] &quot;Eubacterium siraeum et rel.&quot;          </span></span>
<span id="cb16-143"><a href="#cb16-143"></a><span class="co">#&gt;  [7] &quot;Lactococcus&quot;                          </span></span>
<span id="cb16-144"><a href="#cb16-144"></a><span class="co">#&gt;  [8] &quot;Phascolarctobacterium faecium et rel.&quot;</span></span>
<span id="cb16-145"><a href="#cb16-145"></a><span class="co">#&gt;  [9] &quot;Ruminococcus gnavus et rel.&quot;          </span></span>
<span id="cb16-146"><a href="#cb16-146"></a><span class="co">#&gt; [10] &quot;Ruminococcus obeum et rel.&quot;</span></span></code></pre></div>
<div style="page-break-after: always;"></div>
</div>
</div>
<div id="references" class="section level2">
<h2>References</h2>
<p>[1] Ahn S, Datta S. (2023). Differential Co-Abundance Network Analyses for Microbiome Data Adjusted for Clinical Covariates Using Jackknife Pseudo-Values. Under Review at <span class="math inline">\(\textit{BMC Bioinformatics}\)</span>.</p>
<p>[2] McDonald D. et al. (2018). American Gut: an Open Platform for Citizen Science Microbiome Research. <span class="math inline">\(\textit{mSystems}\)</span>. <strong>3</strong>(3), e00031–18</p>
<p>[3] O’Keefe SJ. et al. (2015). Fat, fibre and cancer risk in African Americans and rural Africans. <span class="math inline">\(\textit{Nat Commun}\)</span>. <strong>6</strong>, 6342</p>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
